### ISSUE TITLE
Request Aqara G2H ota 2.2.7

### ISSUE THREAD

**Issue by c20h32o5** [niceboygithub/AqaraGateway#176]
Please, make the dump ota 2.2.7 
Thnks

**Comment by niceboygithub** [comment]
G2H is a bad software design. Have dump rootfs does not helping.

**Comment by c20h32o5** [comment]
I dumped the SPI from the working camera via flashrom and uploaded it to the non-working camera. The camera has loaded, but does not want to bind to the aqara app.

**Comment by niceboygithub** [comment]
You overwrited the factory data.

**Comment by niceboygithub** [comment]
Did you check Wi-Fi driver enabled by wap_supplicant?

**Comment by c20h32o5** [comment]
If you put the factory.ini file on the memory card, then the camera is connected.

**Comment by c20h32o5** [comment]
[   22.924231] ==> rtl8188e_iol_efuse_patch
[   23.664091] IPv6: ADDRCONF(NETDEV_UP): wlan0: link is not ready
[   25.053553] cfg80211: Calling CRDA to update world regulatory domain
[   30.097882] IPv6: ADDRCONF(NETDEV_CHANGE): wlan0: link becomes ready
====================start avahi============================\n
Successfully called chroot().
Starting with address 169.254.7.126
Routable address already assigned, sleeping.
No longer a routable address configured, restarting probe process.
[   48.406420] [Mstar GPIO] gpio(18) to irq(-1)
[   48.429894] [Mstar GPIO] gpio(63) to irq(-1)
Callout BIND, address 169.254.7.126 on interface wlan0
client: ip: either "to" is duplicate, or "scope" is garbage
[   54.205856] cfg80211: Calling CRDA to update world regulatory domain
[   54.614206] ==> rtl8188e_iol_efuse_patch
[   55.069035] MSYS: DMEM request: [VSPL-I1P3B0]:0x00016800
[   55.077852] MSYS: DMEM request: [VSPL-I1P3B1]:0x00016800
[   56.400931] cfg80211: Calling CRDA to update world regulatory domain
Successfully claimed IP address 169.254.7.126
A routable address has been configured.
Callout UNBIND, address 169.254.7.126 on interface wlan0
client: ip: either "to" is duplicate, or "scope" is garbage
client: ip: RTNETLINK answers: Cannot assign requested address
Script execution failed with return value 2

**Comment by niceboygithub** [comment]
It shall be hotspot mode. Did you see it by WiFi?

**Comment by c20h32o5** [comment]
yes. but aqara app says "binding fail". Firmware from another camera, there is a different mac address and probably a serial number.

**Comment by c20h32o5** [comment]
Welcome to Linux.
# agetprop 
[persist.app.box_reset]: [1]
[persist.app.cloud_provisioned]: [false]
[persist.app.coap_token]: [cjYklFTeOZMVtMFc]
[persist.app.holidays]: [1680278400,1680364800,1680624000]
[persist.app.language]: [2]
[persist.app.peerids_0]: [5BH+G0g5KqUzBloVIlR/KSPZqPaoyYd1...]
[persist.app.reboot_count]: [62]
[persist.app.time_zone]: [GMT+03:00]
[persist.app.user_cloud]: [true]
[persist.app.user_paired]: [true]
[persist.app.volume]: [80]
[persist.app.wifi_bssid]: [b0:a7:b9:2a:a4:63]
[persist.app.zigbee_channel]: [25]
[persist.app.zigbee_ieee]: [54ef44100030d643]
[persist.app.zigbee_reboot]: [147]
[persist.sys.coap_key]: [B533D0CFFBD8C82D0C58E4B26466CCBB...]
[persist.sys.did]: [lumi1.54ef44353acc]
[persist.sys.hap_code]: [844-57-690]
[persist.sys.hap_code:]: [488-87-698]
[persist.sys.hap_id]: [QRUK]
[persist.sys.hap_product]: [c43b063b23e2a6b5]
[persist.sys.hap_salt]: [03636132904B5E9C4E07FB7F521E3FEC]
[persist.sys.hap_verifier]: [C8785362C095151B4778D8AD9D81C61D...]
[persist.sys.language]: [us]
[persist.sys.model]: [lumi.camera.gwag03]
[persist.sys.product]: [CH-H01]
[persist.sys.wlan_mac:]: [54:ef:44:35:35:b7]
[persist.sys.wlan_name]: [wlan0]
[persist.sys.zb30_auth]: [true]
[persist.sys.zb_ver]: [0618]
[ro.sys.name]: [Camera-Hub-G2H]
[ro.sys.model]: [lumi.camera.gwagl02]
[ro.sys.product]: [ZNSXJ12LM]
[ro.sys.spu]: [AC004]
[ro.sys.sku]: [000]
[ro.sys.ean13]: [6970504211889]
[ro.sys.manufacturer]: [Aqara]
[ro.sys.vendor]: [Lumi United Technology Co., Ltd.]
[ro.sys.fw_ver]: [2.2.7]
[ro.sys.hw_ver]: [1.0]
[ro.sys.build_num]: [0001]
[ro.sys.acc_tags]: [red]
[ro.sys.prop_port]: [55765]
[ro.sys.prop_valid]: [true]
[sys.acc_serializing]: [false]
[sys.current_state]: [system_idle]


Which of these should you try to change?

**Comment by c20h32o5** [comment]
The camera receives an IP address from the subnet of the router, but after scanning the QR it issues a binding fail. In QR url to aqara website. Perhaps the camera gives certain data when connected.

**Comment by c20h32o5** [comment]
20230331 02:10:50.557 NOTICE   camera_tmp 2537: [AIOT SEND]: {"id":38816,"_to":1,"method":"/v1.0/lumi/dev/connect/gateway","params":{"model":"lumi.camera.gwag03","bindKey":"lumiETptXo5H9tWm","bssid":"b0:a7:b9:2a:a4:63","firmwareVersion":"2.2.7_0001.0618","chipVersion":"mstar","hwMac":"","ipLan":"","longitude":"","latitude":"","ident":"54ef44100030d643"}}
20230331 02:10:50.862 NOTICE   camera_tmp 2402: [AIOT RECV]: {"id":38816,"method":"/v1.0/lumi/dev/connect/gateway","params":{"code":662}}
20230331 02:10:50.862 NOTICE   camera_tmp 2407: [AIOT_RPC_COST]: 1ms
20230331 02:10:50.962 ERROR    camera_tmp 1999: GetAiotToken code: 662
20230331 02:10:50.962 ERROR    camera_tmp 3336: GetAiotToken fail

**Comment by almirus** [comment]
any news?
I have same issue
```
[   16.516544] cfg80211: Calling CRDA to update world regulatory domain
19700101 03:00:15.956 NOTICE   monitor 351: start process /tmp/out/ha_agent...
19700101 03:00:16.457 NOTICE   monitor 351: start process /tmp/out/proxy_server...
19700101 03:00:16.958 NOTICE   monitor 351: start process /tmp/out/camera...
19700101 03:00:17.458 NOTICE   monitor 351: start process /tmp/out/ha_master...
19700101 03:00:17.959 NOTICE   monitor 351: start process /tmp/out/zigbee_agent...
19700101 03:00:18.460 NOTICE   monitor 351: start process /tmp/out/ha_driven...
19700101 03:00:18.961 NOTICE   monitor 368: [monitor] status = 11
19700101 03:00:18.961 NOTICE   monitor 373: child exited abnormal signal number=11
[   21.804167] ==> rtl8188e_iol_efuse_patch
[   28.664169] ==> rtl8188e_iol_efuse_patch
====================start avahi============================\n
Successfully called chroot().
Starting with address 169.254.7.37
[   35.504168] ==> rtl8188e_iol_efuse_patch
Callout BIND, address 169.254.7.37 on interface wlan0
client: ip: either "to" is duplicate, or "scope" is garbage
[   42.354181] ==> rtl8188e_iol_efuse_patch
Successfully claimed IP address 169.254.7.37
[   49.194168] ==> rtl8188e_iol_efuse_patch
[   56.034168] ==> rtl8188e_iol_efuse_patch
[   62.874169] ==> rtl8188e_iol_efuse_patch
[   69.734168] ==> rtl8188e_iol_efuse_patch
[   76.604168] ==> rtl8188e_iol_efuse_patch
[   83.454180] ==> rtl8188e_iol_efuse_patch
[   90.304166] ==> rtl8188e_iol_efuse_patch
[   97.144213] ==> rtl8188e_iol_efuse_patch
[  103.994168] ==> rtl8188e_iol_efuse_patch
[  110.844168] ==> rtl8188e_iol_efuse_patch
[  117.684168] ==> rtl8188e_iol_efuse_patch
[  124.524169] ==> rtl8188e_iol_efuse_patch
[  131.384168] ==> rtl8188e_iol_efuse_patch
[  138.224181] ==> rtl8188e_iol_efuse_patch
[  145.074168] ==> rtl8188e_iol_efuse_patch
[  151.914168] ==> rtl8188e_iol_efuse_patch
[  158.754305] ==> rtl8188e_iol_efuse_patch
#!/bin/sh[  165.604168] ==> rtl8188e_iol_efuse_patch
```

**Comment by almirus** [comment]
do you have 2.2.7_0001.0618 dump?
